src/cargo/ops/cargo_compile/mod.rs

```diff
@@ -476,6 +480,49 @@ pub fn create_bcx<'a, 'gctx>(
             .extend(args);
     }
 
+    {
+        let check_cfg = target_data.check_cfg();
+
+        if check_cfg.exhaustive {
+            let target_cfgs = gctx.target_cfgs()?;
```

Comment:
Is there a reason we are doing this here instead of in our lint phase?

https://github.com/rust-lang/cargo/blob/01e1ab55dd7f391cef809c05a50030ae3bb018d8/src/cargo/core/workspace.rs#L1172-L1249

We could even tie it into our unstable linting system so people can turn it off.

Reply:
I've moved it into to the lint phase/`emit_lints` as suggested. (It's now double gated ;-) `-Zcargo-lints`&`-Zcheck-target-cfgs`) [`cc5b088` (#14581)](https://github.com/rust-lang/cargo/pull/14581/commits/cc5b088b17d0f9c5b564e026bca17f9c579d5850)